store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -9129,22 +9136,28 @@ public void deleteFileSharedWithMe(Mailbox mbox, OperationContext octxt, String
     }
 
     public void updateFileSharedWithMe(Mailbox mbox, OperationContext octxt, String ownerAccountID, int ownerFileId,
-            String remoteUuid, String fileOwnerName, String contentType, String rights, String granteeAccountID)
-            throws ServiceException {
+            String remoteUuid, String fileOwnerName, String contentType, String rights, String granteeAccountID,
+            String fileName, Long date, Long size) throws ServiceException {
         CreateFileSharedWithMe redoRecorder = new CreateFileSharedWithMe(mId, remoteUuid);
         boolean success = false;
         try {
             beginTransaction("updateFileSharedWithMe", octxt, redoRecorder);
             Document item = (Document) mbox.getItemByUuid(remoteUuid + ":" + ownerFileId, Type.DOCUMENT);
             if (item.getOwnerAccountId().equals(ownerAccountID) && item.getAccountId().equals(granteeAccountID)) {
                 Metadata meta = new Metadata();
-                meta.put(Metadata.FN_CREATOR, fileOwnerName);
-                meta.put(Metadata.FN_MIME_TYPE, contentType);
-                meta.put(Metadata.FN_FILE_PERMISSION, rights);
+                meta.put(Metadata.FN_CREATOR, fileOwnerName == "" ? item.creator : fileOwnerName);
```

Comment:
Separate this statement of ternary operators, I would go for one-liner methods may be..

Reply:
created method and return value accordingly

Reply:
reverted the change as discussed for this method